src/ServiceControl/MessageRedirects/Api/MessageRedirectsModule.cs

```diff
@@ -0,0 +1,144 @@
+
+    public class MessageRedirectsModule : BaseModule
+    {
+        public IBus Bus { get; set; }
+
+        public MessageRedirectsModule()
+        {
+
+            Post["/redirects"] = parameters =>
+            {
+                var message = new CreateMessageRedirect
+                {
+                    FromPhysicalAddress = parameters.fromphysicaladdress,
+                    ToPhysicalAddress = parameters.tophysicaladdress
+                };
+
+                if (string.IsNullOrWhiteSpace(message.FromPhysicalAddress) || string.IsNullOrWhiteSpace(message.ToPhysicalAddress))
+                {
+                    return HttpStatusCode.BadRequest;
+                }
+
+                message.MessageRedirectId = DeterministicGuid.MakeId(message.FromPhysicalAddress, message.ToPhysicalAddress);
+
+                using (var session = Store.OpenSession())
+                {
+                    var redirect = session.Load<MessageRedirect>(MessageRedirect.GetDocumentIdFromMessageRedirectId(message.MessageRedirectId));
+
+                    if (redirect != null)
+                    {
+                        return Negotiate.WithReasonPhrase("Duplicate").WithModel(redirect).WithStatusCode(HttpStatusCode.Conflict);
+                    }
+
+                    var dependents = session.Query<MessageRedirect>().Where(r => r.ToPhysicalAddress == message.FromPhysicalAddress).ToList();
+
+                    if (dependents.Any())
+                    {
+                        return Negotiate.WithReasonPhrase("Dependents").WithModel(dependents).WithStatusCode(HttpStatusCode.Conflict);
+                    }
+                }
+
+                Bus.SendLocal(message);
+
+                return HttpStatusCode.Created;
+            };
+
+            Put["/redirects/{messageredirectid}"] = parameters =>
+            {
+                Guid oldMessageRedirectId = parameters.messageredirectid;
+
+                var message = new CreateMessageRedirect
+                {
+                    FromPhysicalAddress = parameters.fromphysicaladdress,
+                    ToPhysicalAddress = parameters.tophysicaladdress
+                };
+
+                if (string.IsNullOrWhiteSpace(message.FromPhysicalAddress) || string.IsNullOrWhiteSpace(message.ToPhysicalAddress))
+                {
+                    return HttpStatusCode.BadRequest;
+                }
+
+                message.MessageRedirectId = DeterministicGuid.MakeId(message.FromPhysicalAddress, message.ToPhysicalAddress);
+
+                Bus.SendLocal(new RemoveMessageRedirect
+                {
+                    MessageRedirectId = oldMessageRedirectId
+                });
+
+                Bus.SendLocal(message);
+
+                return Negotiate.WithStatusCode(HttpStatusCode.Accepted);
+            };
+
+            Delete["/redirects/{messageredirectid}"] = parameters =>
+            {
+
+                Guid messageRedirectId = parameters.messageredirectid;
```

Comment:
We don't check if the input param is null here as we do above? I guess if someone sends us garbage to delete and it's not there then that's OK but we're (Pulse) likely to be the ones that suffer.